src/bootstrap/doc.rs

```diff
@@ -353,7 +353,8 @@ impl Step for Standalone {
         let version_input = builder.src.join("src/doc/version_info.html.template");
         let version_info = out.join("version_info.html");
 
-        if !builder.config.dry_run && !up_to_date(&version_input, &version_info) {
+        if !builder.config.dry_run &&
+            (!version_info.exists() || !up_to_date(&version_input, &version_info)) {
```

Comment:
This change doesn't have any effect as `up_to_date` already checks if the file exists: https://github.com/rust-lang/rust/blob/b7ad3f9fc4b293fd5ada4b975910be26105e5847/src/build_helper/lib.rs#L173-L175

Reply:
Indeed, the check should be on `version_input`. Good catch!

Reply:
That's not what I meant. Checking `version_input` makes even less sense as that file has to exist because it's part of the source code.

Try changing the line:

```rust
up_to_date(&version_info, &html) &&
```

to

```rust
(builder.config.dry_run || up_to_date(&version_info, &html)) &&
```

I think that will fix this issue.